components/proxy/src/main/java/com/hotels/styx/admin/handlers/BackendServiceSerializer.java

```diff
@@ -0,0 +1,165 @@
+import com.hotels.styx.api.service.RewriteConfig;
+import com.hotels.styx.api.service.StickySessionConfig;
+import com.hotels.styx.api.service.TlsSettings;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.base.Objects.toStringHelper;
+/**
+ * Used to provide annotated properties for fields of {@link BackendService} object, required to keep naming of fields
+ * used in configuration.
+ */
+
+public final class BackendServiceSerializer implements Identifiable {
+    private final Id id;
+    private final String path;
+    private final ConnectionPool.Settings connectionPoolSettings;
+    private final Set<Origin> origins;
+    private final HealthCheckConfig healthCheckConfig;
+    private final StickySessionConfig stickySessionConfig;
+    private final List<RewriteConfig> rewrites;
+    private final int responseTimeoutMillis;
+    private final TlsSettings tlsSettings;
+
+    public BackendServiceSerializer(BackendService backendService) {
+        this.id = backendService.id();
+        this.path = backendService.path();
+        this.connectionPoolSettings = backendService.connectionPoolConfig();
+        this.origins = backendService.origins();
+        this.healthCheckConfig = backendService.healthCheckConfig();
+        this.stickySessionConfig = backendService.stickySessionConfig();
+        this.rewrites = backendService.rewrites();
+        this.responseTimeoutMillis = backendService.responseTimeoutMillis();
+        this.tlsSettings = backendService.tlsSettings().orElse(null);
+    }
+
+    @Override
+    public Id id() {
+        return this.id;
+    }
+
+    @JsonProperty("id")
+    String idAsString() {
+        return this.id.toString();
+    }
+
+    @JsonProperty("path")
+    public String path() {
+        return this.path;
+    }
+
+    @JsonProperty("origins")
+    public Set<Origin> origins() {
+        return this.origins;
+    }
+
+    @JsonProperty("connectionPool")
+    public ConnectionPool.Settings connectionPoolConfig() {
+        return this.connectionPoolSettings;
+    }
+
+    @JsonProperty("healthCheck")
+    public HealthCheckConfig healthCheckConfig() {
+        return healthCheckConfig;
+    }
+
+    @JsonProperty("stickySession")
+    public StickySessionConfig stickySessionConfig() {
+        return this.stickySessionConfig;
+    }
+
+    @JsonProperty("rewrites")
+    public List<RewriteConfig> rewrites() {
+        return this.rewrites;
+    }
+
+    @JsonProperty("responseTimeoutMillis")
+    public int responseTimeoutMillis() {
+        return this.responseTimeoutMillis;
+    }
+
+    public Optional<TlsSettings> tlsSettings() {
+        return Optional.ofNullable(this.tlsSettings);
+    }
+
+    @JsonProperty("tlsSettings")
+    private TlsSettings getTlsSettings() {
+        return this.tlsSettings().orElse(null);
+    }
+
+    public BackendService.Protocol protocol() {
+        if (tlsSettings == null) {
```

Comment:
Using static-imports, this code could be `return tlsSettings == null ? HTTP : HTTPS;`